guidelines/GOLANG.md

```diff
@@ -0,0 +1,31 @@
+Golang guidelines
+=====================
+
+These are the Digital Publishing guidelines for developing code using Golang. They can also be used as a code review
+checklist.
+
+#### General
+* Start with the language level recommendations for Golang in [Effective Go](https://golang.org/doc/effective_go.html) 
+and the [code review guidelines](https://github.com/golang/go/wiki/CodeReviewComments)
+* [Listen for OS signals](https://golang.org/pkg/os/signal/) and gracefully shutdown the app when one is received
+
+#### Tooling
+
+* Use [gometalinter](https://github.com/alecthomas/gometalinter) to identify code issues
+* Prefer using [GoConvey](https://github.com/smartystreets/goconvey) as a test tool
+* Use [govendor](https://github.com/kardianos/govendor) for vendoring dependencies
+
+#### Libraries
+
+* Utilise the [go-ns](https://github.com/ONSdigital/go-ns) library for common functionality. Extract common code from projects and contribute it to go-ns
+    * Use the common logging implementation
+    * Default to the go-ns HTTP server implementation
+        * If you don't, ensure your server utilises the common middleware handlers
+            * timeout
```

Comment:
we should probably remove this, the timeout handler provided in go-ns is broken and should be removed

Reply:
Removed the timeout handler from the list. I will remove the actual handler from go-ns.